python/lsst/afw/math/background/backgroundContinued.py

```diff
@@ -19,14 +19,54 @@
 # You should have received a copy of the GNU General Public License
 # along with this program.  If not, see <https://www.gnu.org/licenses/>.
 
+import warnings
+
 from lsst.utils import continueClass
-from .background import Background
+from lsst.utils.deprecated import deprecate_pybind11
+from .background import Background, BackgroundControl, BackgroundMI
+from ..interpolate import Interpolate
 
 __all__ = []  # import this module only for its side effects
 
 
+@continueClass  # noqa: F811
+class BackgroundControl:
+    _init = BackgroundControl.__init__
+
+    def __init__(self, *args, **kwargs):
+        # This constructor called dozens of times; warn only for invalid use
+        if (args and (isinstance(args[0], str) or isinstance(args[0], Interpolate.Style))) \
+                or "style" in kwargs:
+            warnings.warn('Call to deprecated method __init__. (Overloads that take a ``style`` parameter '
```

Comment:
I did do this in ip_diffim DM-21065 when I wanted to only issue a warning when certain parameters were given.